src/Chapter3.hs

```diff
@@ -971,6 +1092,42 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday
+  deriving (Show, Eq, Bounded, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend day
+  | day == pred maxBound  = True
+  | day == maxBound       = True
+  | otherwise             = False
+
+{-
+-- I would like to be able to express the cycling property of the function nextDay in terms of its minBound and maxBound
+-- I've had no luck so far...
+nextDay :: Weekday -> Weekday
+nextDay maxBound = minBound
+nextDay x = succ x
+-}
+
+-- I'm not quite happy with this either because I would have loved to use succ. Oh well w/e.
+nextDay :: Weekday -> Weekday
+nextDay x = cycle [minBound .. maxBound] !! (fromEnum x + 1)
+
+{-
+-- Actually maybe it's better to do a tail recursion here
+daysToParty :: Weekday -> Int
+daysToParty Friday  = 0
+daysToParty x       = 1 + daysToParty (nextDay x)
+-}
+
+-- Am I using a bazooka to shoot a fly here?
+daysToParty :: Weekday -> Int
+daysToParty = go 0
+  where
+    go :: Int -> Weekday -> Int
+    go n Friday = n
+    go n x = go (n + 1) (nextDay x)
```

Comment:
You can write even better implementation using `fromEnum` and `mod`, especially if you remember how `mod` works for negative numbers from Chapter 1 🙂